### PR TITLE
Fix JWT Policy fields

### PR DESCRIPTION
**Pull request by didierofrivia**
**What this PR does / why we need it**:

This update supports new json schema options that are required by some policies (I.E: JWT)

**Which issue(s) this PR fixes** 

Fixes https://issues.jboss.org/browse/THREESCALE-3509

**Verification steps** 
* Chose a Product/API
* Visit Integration -> Policies
* Add JWT policy and configure its options.

It should work as expected

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by didierofrivia** [comment]
I haven't experienced https://issues.jboss.org/browse/THREESCALE-3475 happening after this upgrade. I've deployed in preview env and checked it locally. cc: @thomasmaas @josemigallas @damianpm

**Comment by thomasmaas** [comment]
@josemigallas, the last time we tried to fix this issue something went wrong in production. Do you remember what happened?

**Comment by josemigallas** [comment]
> @josemigallas, the last time we tried to fix this issue something went wrong in production. Do you remember what happened?

Last time, version 1.8.0 was installed by NPM without us knowing, leading to an uncontrolled state. Here's the relevant link: https://github.com/3scale/porta/pull/1194

As per the concrete error I can't remember but it was on JS side. By verifying it runs in local/preview it should be safe to ship it.

File: app/javascript/src/Policies/components/PolicyConfig.jsx [hunk 1]
```diff
@@ -18,13 +18,66 @@ type Props = {
...
+      </header>
+      <h2 className="PolicyConfiguration-name">{humanName}</h2>
+      <p className="PolicyConfiguration-version-and-summary">
+        <span className="PolicyConfiguration-version">{version}</span>
+        {' - '}
+        <span className="PolicyConfiguration-summary">{summary}</span>
+      </p>
+      <p className="PolicyConfiguration-description">{description}</p>
+      {isPolicyVisible &&
+      <label className="Policy-status" htmlFor="policy-enabled">
```
Comment (josemigallas) [inline app/javascript/src/Policies/components/PolicyConfig.jsx]:
```suggestion
        <label className="Policy-status" htmlFor="policy-enabled">
```
I think indentation is a bit short on lines 53, 63 and 73 right after the `... &&` predicate.

File: app/javascript/src/Policies/components/PolicyConfig.jsx [hunk 2]
```diff
@@ -18,13 +18,66 @@ type Props = {
...
+        onSubmit={onSubmit(policy)}
+      >
+        <button className='btn btn-info' type="submit">Update Policy</button>
+      </Form>
+      }
+      {removable &&
+      <div
+        className="PolicyConfiguration-remove btn btn-danger btn-sm"
+        onClick={remove}>
+        <i className="fa fa-trash"> </i> Remove
```
Comment (josemigallas) [inline app/javascript/src/Policies/components/PolicyConfig.jsx]:
```suggestion
        <i className="fa fa-trash"> Remove
```

File: app/javascript/src/Policies/components/PolicyConfig.jsx [hunk 3]
```diff
@@ -18,13 +18,66 @@ type Props = {
...
+        formData={data}
+        onSubmit={onSubmit(policy)}
+      >
+        <button className='btn btn-info' type="submit">Update Policy</button>
+      </Form>
+      }
+      {removable &&
+      <div
+        className="PolicyConfiguration-remove btn btn-danger btn-sm"
+        onClick={remove}>
```
Comment (josemigallas) [inline app/javascript/src/Policies/components/PolicyConfig.jsx]:
```suggestion
        onClick={remove}
      >
```

File: app/javascript/src/Policies/components/PolicyConfig.jsx [hunk 4]
```diff
@@ -18,13 +18,66 @@ type Props = {
...
+      </label>
+      }
+      {isPolicyVisible &&
+      <Form
+        className="PolicyConfiguration-form"
+        schema={configuration}
+        formData={data}
+        onSubmit={onSubmit(policy)}
+      >
+        <button className='btn btn-info' type="submit">Update Policy</button>
```
Comment (josemigallas) [inline app/javascript/src/Policies/components/PolicyConfig.jsx]:
```suggestion
        <button className="btn btn-info" type="submit">Update Policy</button>
```

File: app/javascript/src/Policies/components/PolicyConfig.jsx [hunk 5]
```diff
@@ -18,13 +18,66 @@ type Props = {
...
-    removePolicy={actions.removePolicyFromChain}
-    closePolicyConfig={actions.closePolicyConfig}
-    updatePolicy={actions.updatePolicyConfig}
-  />)
+  const { submitPolicyConfig, removePolicyFromChain, closePolicyConfig, updatePolicyConfig } = actions
+  const { humanName, version, summary, description, enabled, configuration, data, removable } = policy
+
+  const onSubmit = (policy) => {
+    return ({formData, schema}) => {
+      submitPolicyConfig({...policy, ...{configuration: schema, data: formData}})
```
Comment (josemigallas) [inline app/javascript/src/Policies/components/PolicyConfig.jsx]:
```suggestion
      submitPolicyConfig({...policy, ...configuration: schema, data: formData})
```

File: app/javascript/src/Policies/components/PolicyConfig.jsx [hunk 6]
```diff
@@ -18,13 +18,66 @@ type Props = {
...
+    }
+  }
+  const togglePolicy = (event) => {
+    updatePolicyConfig({...policy, ...{enabled: event.target.checked}})
```
Comment (josemigallas) [inline app/javascript/src/Policies/components/PolicyConfig.jsx]:
```suggestion
    updatePolicyConfig({...policy, enabled: event.target.checked})
```

File: app/javascript/src/Policies/components/PolicyConfig.jsx [hunk 7]
```diff
@@ -18,13 +18,66 @@ type Props = {
...
+  const remove = () => removePolicyFromChain(policy)
+  const cancel = () => closePolicyConfig()
+
+  const isPolicyVisible = isNotApicastPolicy(policy)
+
+  return (
+    <section className="PolicyConfiguration">
+      <header>
+        <h2>Edit Policy</h2>
+        <HeaderButton type='cancel' onClick={cancel}>
```
Comment (josemigallas) [inline app/javascript/src/Policies/components/PolicyConfig.jsx]:
```suggestion
        <HeaderButton type="cancel" onClick={cancel}>
```

File: app/javascript/src/Policies/components/PolicyConfig.jsx [hunk 8]
```diff
@@ -18,13 +18,66 @@ type Props = {
...
+      <p className="PolicyConfiguration-version-and-summary">
+        <span className="PolicyConfiguration-version">{version}</span>
+        {' - '}
+        <span className="PolicyConfiguration-summary">{summary}</span>
+      </p>
+      <p className="PolicyConfiguration-description">{description}</p>
+      {isPolicyVisible &&
+      <label className="Policy-status" htmlFor="policy-enabled">
+        <input
+          id="policy-enabled" name="policy-enabled" type="checkbox"
```
Comment (josemigallas) [inline app/javascript/src/Policies/components/PolicyConfig.jsx]:
```suggestion
          id="policy-enabled"
          name="policy-enabled"
          type="checkbox"
```

**Pull request merged**